crates/hotshot/task-impls/src/helpers.rs

```diff
@@ -222,16 +222,20 @@ async fn decide_epoch_root<TYPES: NodeType, I: NodeImplementation<TYPES>>(
         tracing::info!("Time taken to store epoch root: {:?}", start.elapsed());
 
         start = Instant::now();
-        let write_callback = {
+
+        let add_epoch_root_worker = {
             tracing::debug!("Calling add_epoch_root for epoch {next_epoch_number}");
             let membership_reader = membership.read().await;
-            membership_reader
-                .add_epoch_root(next_epoch_number, decided_leaf.block_header().clone())
-                .await
+            membership_reader.add_epoch_root(next_epoch_number, decided_leaf.block_header().clone())
         };
-        if let Some(write_callback) = write_callback {
-            let mut membership_writer = membership.write().await;
-            write_callback(&mut *membership_writer);
+
+        if let Some(worker) = add_epoch_root_worker {
+            let add_epoch_root_updater = worker().await;
```

Comment:
This still isn't moving this off the critical path.  We aren't holding the membership read lock anymore but it's still going to block the consensus task until `add_epoch_root_updater` completes.  The idea I had was that removing the requirement to lock the membership would allow us to `tokio::spawn` this and thus it consensus can continue while the stake table stuff happens in the background